Makefile

```diff
@@ -3,13 +3,49 @@
 SOURCES = AddressResolver.cpp Profile.cpp
 HEADERS = AddressResolver.h Profile.h
 
+KCACHEGRIND = kcachegrind
+
+CFLAGS = -Wall -Wextra -Werror -g -fno-omit-frame-pointer
+
 all: pgcollect pginfo pgconvert
 
 pgcollect: pgcollect.c
-	$(CC) -std=gnu99 -Wall -Wextra -g -D_GNU_SOURCE -o pgcollect pgcollect.c ${FLAGS}
+	$(CC) -std=gnu99  $(CFLAGS) -D_GNU_SOURCE -o pgcollect pgcollect.c   ${FLAGS}
 
 pginfo: pginfo.cpp $(SOURCES) $(HEADERS)
-	$(CXX) -std=c++03 -Wall -Wextra -g -o pginfo pginfo.cpp $(SOURCES) -ldw -lelf ${FLAGS}
+	$(CXX) -std=c++03 $(CFLAGS)               -o pginfo    pginfo.cpp    $(SOURCES) -ldw -lelf ${FLAGS}
 
 pgconvert: pgconvert.cpp $(SOURCES) $(HEADERS)
-	$(CXX) -std=c++03 -Wall -Wextra -g -o pgconvert pgconvert.cpp $(SOURCES) -ldw -lelf ${FLAGS}
+	$(CXX) -std=c++03 $(CFLAGS)               -o pgconvert pgconvert.cpp $(SOURCES) -ldw -lelf ${FLAGS}
+
+.PHONY: clean clean-dev clean-check
+
+clean: clean-dev clean-check
+	rm -rf pgcollect pginfo pgconvert
+
+clean-dev:
+	rm -rf *.o
+
+clean-check:
+	rm -rf *check*
+
+callgrind.out.check_ls.pgdata callgrind.out.check.pgdata:
+	$(MAKE) check
+
+check:	pgcollect pgconvert pginfo
+	@echo ""; echo "collecting some data of ls binary (likely without full symbols)..."
+	./pgcollect check_ls.pgdata -F 8192 -s -- ls -l /usr/bin  1>/dev/null
```

Comment:
I'd rather omit -F 8192, this is quite high frequency, which can be blocked.

Reply:
The issue is that we do want a bunch of events to gather. If we take a process that takes longer and is "quite guaranteed" to execute we can a lower frequency.
Any suggestion? And what frequency would you use for that?

And just for me as I see something to learn: what do you mean with "blocked"? How would this be configured?

Reply:
There is a bunch of kernel tunables under `/proc/sys/kernel/perf_event_*`. `perf_event_max_sample_rate` can block high frequency.

Reply:
From what I've found 2048 _should_ work in most cases, but in any case I'll put that into PGCOLLECT_FLAGS so both the frequency and the software event collection can be adjusted easily on the make invocation.